src/Okta.Idx.Sdk/IdxClient.cs

```diff
@@ -1510,5 +1536,240 @@ public async Task<TResponse> SendAsync<TResponse>(HttpRequest request, HttpVerb
+                {
+                    return new PollResponse
+                    {
+                        IdxContext = idxContext,
+                        AuthenticationStatus = AuthenticationStatus.AwaitingChallengeAuthenticatorSelection,
+                        Authenticators = IdxResponseHelper.ConvertToAuthenticators(introspectResponse.Authenticators?.Value, introspectResponse.AuthenticatorEnrollments?.Value),
+                        CanSkip = introspectResponse.ContainsRemediationOption(RemediationType.Skip),
+                    };
+                }
+
+                if (introspectResponse.ContainsRemediationOption(RemediationType.SelectAuthenticatorEnroll))
+                {
+                    return new PollResponse
+                    {
+                        IdxContext = idxContext,
+                        AuthenticationStatus = AuthenticationStatus.AwaitingAuthenticatorEnrollment,
+                        Authenticators = IdxResponseHelper.ConvertToAuthenticators(introspectResponse.Authenticators?.Value),
+                        CanSkip = introspectResponse.ContainsRemediationOption(RemediationType.Skip),
+                    };
+                }
+            }
+
+            throw new UnexpectedRemediationException(
+                new List<string>
+                {
+                    RemediationType.EnrollPoll,
+                    RemediationType.SelectAuthenticatorAuthenticate,
+                    RemediationType.SelectAuthenticatorEnroll,
+                },
+                introspectResponse);
+        }
+
+        /// <inheritdoc/>
+        public async Task<PollResponse> PollAuthenticatorPushStatusAsync(IIdxContext idxContext, CancellationToken cancellationToken = default)
+        {
+            var introspectResponse = await IntrospectAsync(idxContext, cancellationToken);
+
+            IdxRequestPayload requestPayload = new IdxRequestPayload
+            {
+                StateHandle = introspectResponse.StateHandle,
+            };
+
+            if (!introspectResponse.ContainsRemediationOption(RemediationType.ChallengePoll))
+            {
+                throw new UnexpectedRemediationException(RemediationType.ChallengePoll, introspectResponse);
+            }
+
+            var challengeResponse = await introspectResponse.ProceedWithRemediationOptionAsync(RemediationType.ChallengePoll, requestPayload, cancellationToken);
+            bool continuePolling = challengeResponse.ContainsRemediationOption(RemediationType.ChallengePoll, out IRemediationOption challengePollRemediationOption);
+
+            if (continuePolling)
+            {
+                return new PollResponse
+                {
+                    AuthenticationStatus = AuthenticationStatus.AwaitingAuthenticatorVerification,
+                    ContinuePolling = continuePolling,
+                    Refresh = challengePollRemediationOption?.Refresh ?? 4000,

```

Comment:
Since 4000 is being set multiple times, let's create a constant and use that instead. Also, why are we setting this up to 4000?

Reply:
We should indicate in the comments if this is milliseconds or something else.